commons/all/src/main/java/org/infinispan/commons/dataconversion/RFC4648Codec.java

```diff
@@ -0,0 +1,93 @@
+package org.infinispan.commons.dataconversion;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.infinispan.commons.dataconversion.MediaType.APPLICATION_OBJECT;
+
+import java.util.Base64;
+import java.util.Optional;
+
+import org.infinispan.commons.logging.Log;
+import org.infinispan.commons.logging.LogFactory;
+
+/**
+ * Handles base16 and base64 data encodings as specified in the <a href="https://www.ietf.org/rfc/rfc4648.txt">RFC 4648</a>.
+ *
+ * It used when the {@link MediaType} contains an <code>encoding</code> param, e.g.
+ * <br/>
+ * <code>application/octet-stream; encoding=hex</code>.
+ * <br>
+ * Valid encodings are "hex" (base16) and "base64".
+ *
+ * @since 13.0
+ */
+class RFC4648Codec implements MediaTypeCodec {
+   private static final Log log = LogFactory.getLog(RFC4648Codec.class);
+
+   @Override
+   public Object decodeContent(Object content, MediaType contentType) {
+      Optional<String> optionalEncoding = contentType.getParameter("encoding");
+      if (!optionalEncoding.isPresent()) {
+         if (content instanceof String) {
+            if (contentType.match(MediaType.APPLICATION_OCTET_STREAM) || (contentType.match(APPLICATION_OBJECT) && contentType.getClassType() != null && contentType.getClassType().equals("ByteArray"))) {
```

Comment:
you could add the "content instance of String && " the test stuff to avoid the 3 ifs